sycl/include/sycl/detail/pi.h

```diff
@@ -2027,6 +2027,16 @@ __SYCL_EXPORT pi_result piTearDown(void *PluginParameter);
 /// timestamp
 __SYCL_EXPORT pi_result piPluginGetLastError(char **message);
 
+/// API to get backend specific optimization option.
+/// \param opt_level is an integer that contains frontend optimization level.
+/// \param backend_option is used to return the backend optimization option
+/// corresponding to frontend optimization level.
+///
+/// \return PI_SUCCESS is returned always. If a valid option is not
+/// available, an empty string is returned.
+__SYCL_EXPORT pi_result
+piPluginGetBackendOptimizationOption(int opt_level, char **backend_option);
```

Comment:
New API entry might be relevant for the UR team, cc: @kbenzie

Reply:
UR doesn't have an analagous concept to the plugin entry points, could this be platform instead?

Reply:
> could this be platform instead?

+1

Reply:
Hi @jchlanda, @kbenzie and @smaslov-intel 

Can you please elaborate on the change required here?

Thanks

Reply:
Essentially the entry point would become:

```cpp
__SYCL_EXPORT pi_result
piPlatformGetBackendOptimizationOption(pi_platform platform,
                                       int opt_level, char **backend_option);
```

There are a couple of reasons:

* Encapsulation of state within the platform object.
* Remove the need to have special handling in the UR loader, instead the dispatch table for platform entry points would be used.

I've not delved into how the implementation all fits togetherm hopefully this is feasible?

Reply:
We are tracking this change with https://github.com/oneapi-src/unified-runtime/issues/427